### PR TITLE
Misc async methods tinkerings

### PR DESCRIPTION
**Pull request by dvdplm**
I was late to review #327 so here's a PR with suggestions I wanted to make there.﻿

It's a bit noisy mainly because of a bunch of renames:

`Methods` –> `SyncMethods`
`MethodsHandler` –> `Methods`

…and all the fallout from that.

### CODE REVIEW

File: ws-server/src/server.rs [hunk 1]
```diff
@@ -124,40 +124,49 @@ async fn background_task(
...
+	// Note: This handler expects method existence to be checked prior to starting the server and will panic if the
 	// method does not exist.
 	let execute_async = |tx: MethodSink, req: OwnedJsonRpcRequest| {
 		let async_methods = methods.clone();
 		async move {
 			let req = req.borrowed();
-			let method = async_methods.async_method(&*req.method).unwrap();
+			let method = async_methods
+				.async_method(&*req.method)
+				.expect(&format!("async method '{}' is not registered on the server – this is a bug", req.method));
```
Comment (popzxc) [inline ws-server/src/server.rs:161]: It's generally better to use `unwrap_or_else(|| panic!(..))` in such cases.

**Review by popzxc — APPROVED** [review]
Totally makes sense.

File: types/src/v2/params.rs [hunk 2]
```diff
@@ -209,7 +209,7 @@ impl<'a> Id<'a> {
 	}
 }
 
-/// Owned version of [`Id`] that always allocates memory for its members.
+/// Owned version of [`Id`] that allocates memory for the `Number` and `Str` variants.
```
Comment (niklasad1) [inline types/src/v2/params.rs:212]: should only allocate for `Str` FWIW

File: utils/src/server/rpc_module.rs [hunk 3]
```diff
@@ -283,11 +283,11 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 	}
 
 	/// Convert a module into methods. Consumes self.
-	pub fn into_methods(self) -> MethodsHolder {
+	pub fn into_methods(self) -> Methods {
 		self.methods
 	}
 
-	/// Merge two [`RpcModule`]'s by adding all [`Method`]s from `other` into `self`.
+	/// Merge two [`RpcModule`]'s by adding all [`SyncMethod`]s from `other` into `self`.
```
Comment (niklasad1) [inline utils/src/server/rpc_module.rs:290]:
```suggestion
	/// Merge two [`RpcModule`]'s by adding all [`Method`]s from `other` into `self`.
```

File: ws-server/src/server.rs [hunk 4]
```diff
@@ -124,40 +124,49 @@ async fn background_task(
...
 	//
-	// Note: This handler expects method existence to be checked prior to the call and will panic if
+	// Note: This handler expects method existence to be checked prior to starting the server and will panic if the
 	// method does not exist.
 	let sync_methods = methods.clone();
 	let execute_sync = move |tx: &MethodSink, req: JsonRpcRequest| {
-		let method = sync_methods.method(&*req.method).unwrap();
+		let method = sync_methods
+			.sync_method(&*req.method)
+			.expect(&format!("sync method '{}' is not registered on the server – this is a bug", req.method));
```
Comment (niklasad1) [inline ws-server/src/server.rs:140]: unwrap_or_else(|| panic!("sync method '{}' is not registered on the server – this is a bug", req.method));

File: types/src/v2/params.rs (hunk 2 above)
Comment (niklasad1) [inline types/src/v2/params.rs:212]:
```suggestion
/// Owned version of [`Id`] that allocates memory for the `Str` variant.
```

**Review by niklasad1 — CHANGES_REQUESTED** [review]
great, some minor things to clean up before merging

see inline comments

**Review by maciejhirsz — APPROVED** [review]

**Review by niklasad1 — APPROVED** [review]

**Pull request merged**